File: Locator/src/Locator.Contracts/Vacancies/Responses/VacanciesResponse.cs
using Locator.Contracts.Vacancies.Dto;

namespace Locator.Contracts.Vacancies.Responses;

public record VacanciesResponse(
    long Count, 
    IEnumerable<FullVacancyDto> Vacancies, 
    int Page, 
    int Pages, 
    int PerPage);

File: client/src/contracts/responses/VacanciesResponse.ts
import type { FullVacancyDto } from '../dtos/FullVacancyDto';

export interface VacanciesResponse {
    vacancies: FullVacancyDto[];
    pages: number;
}

export interface UnauthorizedResponse {
    unauthorized: true;
}

export type VacanciesFetchResult = VacanciesResponse | UnauthorizedResponse | null;